blessed/src/main/java/com/welie/blessed/BluetoothPeripheralManager.kt

```diff
@@ -79,10 +83,9 @@ class BluetoothPeripheralManager(private val context: Context, private val bluet
 
                     // First check if this is a connecting peripheral because Android
                     // will also call this callback for connecting peripherals
-                    centralManager?.let {
-                        if (it.unconnectedPeripherals.containsKey(device.address)) {
-                            return
-                        }
+                    if (peripheralQualifier?.isDeviceAPeripheral(device) == true) {
+                        Logger.v(TAG, "Device ${device.name} is a peripheral, ignoring STATE_CONNECTED event")
```

Comment:
Sometimes name is not set, maybe it's worth adding the MAC address of the device to the log as well? WDYT?

Reply:
Sure 👍